### DISCUSSION TITLE
Server Restart Required - what services could be restarted without rebooting the server?

### DISCUSSION THREAD

**Discussion by compumatter** [inventree/InvenTree#4216] [General]
Hello.
I'm getting the following message.
![image](https://user-images.githubusercontent.com/121611/212604620-4aad0249-9578-415d-8af4-f5f8a7dad1e0.png)
This server has other software running in addition to Inventree.  I would like not to reboot the entire server each time this comes up.

So I'm wondering what processes / services need to be restarted to properly satisfy this requirement.

Thanks.

**Comment by wolflu05** [comment]
Hello @compumatter ,

That depends on how you deployed inventree. Please give us more information about that.

**Reply by compumatter** [reply]
Thanks @wolflu05 I've gone ahead and created a short video showing exactly how I installed Inventree.   It is a bare metal install on Ubuntu 22.04.  This video may help others who might want to use Ansible to install InvenTree as well.  

https://www.youtube.com/watch?v=vyCu_1zf2HQ

**Reply by wolflu05** [reply]
You can either use the `supervisorctl restart inventree-server inventree-cluster` command, or connect to port 9001 on your server via a Webbrowser as the [documentation](https://docs.inventree.org/en/latest/start/production/#view-process-status) suggests.

**Reply by matmair** [reply]
In general installs on bare metal are easier handled with the native packages as those integrate natively into the native process/service manager

**Reply by compumatter** [reply]
@wolflu05 thank you.  That will be much better than a server restart.  Question about that port 9001.  I have found no reference to that anywhere.  If you know where that is, please drop me a link.  I'd like to see it.  I'm using port 8000 now, which I seem to find references to everywhere.  So your recommendation is to change 8000 > 9001 ?

**Reply by compumatter** [reply]
@matmair I would prefer to go native.  I believe this page is what you're talking about: https://docs.inventree.org/en/latest/start/installer  

The only reason I didn't follow along as it specifically says Debian 11 / Ubuntu 20.04.   I'm using Ubuntu 22.04 so I kept looking for a solution.  Do those docs now apply to 22.04?  That would be a lot easier than what I went through to get it installed.   

I ended up finding this document https://docs.inventree.org/en/latest/start/install/ which did not specifically require 20.04 so I dug in on it.

If that applies to 22.04 I'll uninstall and give that a go.  If it does apply, can I export the existing database and import into the package maintained version?

**Reply by eeintech** [reply]
@compumatter I just gave it a go and no problem to run on 22.04. It installs `0.9.0` so still need update afterwards.

**Reply by matmair** [reply]
@compumatter It runs on 22.04 but we need to ship our own OpenSSL dependency as 22.04 switched to libopenssl3 while we need libopenssl1.1
I deploy 22.04 with the installer script. Maybe this can be an inspiration:
https://github.com/invenhost/inventree_droplet/blob/master/files/var/lib/cloud/scripts/per-instance/001_onboot

**Reply by compumatter** [reply]
@eeintech I just ran the script wget -N https://raw.githubusercontent.com/InvenTree/InvenTree/master/contrib/install.sh && bash install.sh and runs without error.  I find the Inventree software with /opt/inventree and it appears they are using port 6000 as the default.

I can see gunicorn services are running on port 6000 but I get a complete fail in the browser at http://my.ip.ad.rs:6000
![image](https://user-images.githubusercontent.com/121611/213021982-fd3739e8-cd76-4cc4-8545-1f854bb95fb0.png)

ps aux | grep inventree reveals the following:

` ps aux | grep inventr
inventr+  491602 39.2  0.2  62956 47116 ?        R    14:04   0:01 /opt/inventree/env/bin/python3 env/bin/gunicorn --chdir /opt/inventree/InvenTree -c InvenTree/gunicorn.conf.py InvenTree.wsgi -b 0.0.0.0:6000
inventr+  491626 49.0  0.1  43264 31284 ?        Rs   14:04   0:00 env/bin/python InvenTree/manage.py qcluster
`
Any thoughts on that?

**Reply by eeintech** [reply]
At the end of the script, what did it say to you? For me it just ran on `http://localhost/`:

``` shell
# Starting InvenTree
####################################################################################
This InvenTree install uses nginx, the settings for the webserver can be found in
/etc/nginx/sites-enabled/inventree.conf
Try opening InvenTree with either
http://localhost/ or http://184.95.xx.50/

Admin user data:
   Email: admin@example.com
   Username: admin
   Password: abcdefghijklmnopqrstuvwxyz
####################################################################################
```

**Comment by compumatter** [comment]
Never saw that feed back.  Could you send me. Link to the very script you ran so I know I'm in the same page ThanksSincerely,JayJay LeporeCompuMatterOffice: 360 299 2428Cell: 360 202 ***@***.*** you are satisfied with your service, please consider providing us a social media review at https://compumatter.com/reviews...If you are not, please let me know so I can make it betterOn Jan 18, 2023, at 6:48 AM, Francois ***@***.***> wrote:﻿
At the end of the script, what did it say to you? For me it just ran on http://localhost/:
# Starting InvenTree
####################################################################################
This InvenTree install uses nginx, the settings for the webserver can be found in
/etc/nginx/sites-enabled/inventree.conf
Try opening InvenTree with either
http://localhost/ or http://184.95.212.50/

Admin user data:
   Email: ***@***.***
   Username: admin
   Password: abcdefghijklmnopqrstuvwxyz
####################################################################################

**Reply by eeintech** [reply]
I ran the exact same command you did: https://docs.inventree.org/en/latest/start/intro/
Therefore make sure to check the output of the script, it is likely that it failed and you need to find out why.
